flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -194,44 +166,35 @@ Fortran::lower::CharacterExprHelper::toExtendedValue(mlir::Value character,
   return fir::CharBoxValue{base, resultLen};
 }
 
-/// Get canonical `!fir.ref<!fir.char<kind>>` type.
+/// Get `!fir.char<kind, ?>` type with the same kind as \p str type.
 mlir::Type
-Fortran::lower::CharacterExprHelper::getReferenceType(mlir::Value str) const {
-  return builder.getRefType(getCharacterType(str));
+Fortran::lower::CharacterExprHelper::getUnknownLenType(mlir::Value str) const {
```

Comment:
This seems like it would be useful enough to make it a first-class `static get` method (`getUnknownLen`?) directly in `CharacterType`. Putting it there has the added bonus of making it easy to find.

Reply:
Added `fir::CharacterType::getUnkownLen(ctxt, kind)` and removed this one.